cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -91,6 +103,150 @@ func stampedAction(stamp string, action func() error) error {
 	return ioutil.WriteFile(stampFile, nil, 0644)
 }
 
+// SystemdMountOptions reflects the set of options for mounting something using
+// systemd-mount(1)
+type SystemdMountOptions struct {
+	// is tmpfs indicates that "what" should be ignored and a new tmpfs should
+	// be mounted at the location
+	IsTmpfs bool
+	// SurvivesPivotRoot indicates that the mount should persist from the
+	// initramfs to after the pivot_root to normal userspace
+	// this is done by creating systemd unit .wants symlinks on
+	// initrd-device.target in /run
+	// TODO: all current instances of doSystemdMount set this to true, should we
+	// invert or just drop the option and make it the default behavior ?
+	SurvivesPivotRoot bool
+	// NeedsFsck indicates that before returning to the caller, an fsck check
+	// should be performed on the thing being mounted
+	NeedsFsck bool
+	// NoWait will not wait until the systemd unit is active and running, which
+	// is the default behavior
+	NoWait bool
+}
+
+var (
+	doSystemdMount = doSystemdMountImpl
+	counter        = 1
+)
+
+// doSystemdMount will mount "what" at "where" using systemd-mount(1) with
+// various options
+func doSystemdMountImpl(what, where string, opts *SystemdMountOptions) error {
```

Comment:
this probably should leave in a separate .go file with the struct and with its own testing